pkg/remote/oci.go

```diff
@@ -0,0 +1,135 @@
+/*
+   Copyright 2020 Docker Compose CLI authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package remote
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/adrg/xdg"
+	"github.com/distribution/distribution/v3/reference"
+	"github.com/docker/buildx/store/storeutil"
+	"github.com/docker/buildx/util/imagetools"
+	"github.com/docker/cli/cli/command"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+
+	"github.com/compose-spec/compose-go/loader"
+	"github.com/pkg/errors"
+)
+
+func OCIRemoteLoaderEnabled() (bool, error) {
+	if v := os.Getenv("COMPOSE_EXPERIMENTAL_OCI_REMOTE"); v != "" {
+		enabled, err := strconv.ParseBool(v)
+		if err != nil {
+			return false, errors.Wrap(err, "COMPOSE_EXPERIMENTAL_OCI_REMOTE environment variable expects boolean value")
+		}
+		return enabled, err
+	}
+	return false, nil
+}
+
+func NewOCIRemoteLoader(dockerCli command.Cli) (loader.ResourceLoader, error) {
+	// xdg.CacheFile creates the parent directories for the target file path
+	// and returns the fully qualified path, so use "git" as a filename and
+	// then chop it off after, i.e. no ~/.cache/docker-compose/git file will
+	// ever be created
+	cache, err := xdg.CacheFile(filepath.Join("docker-compose", "oci"))
+	if err != nil {
+		return nil, fmt.Errorf("initializing git cache: %w", err)
+	}
+	cache = filepath.Dir(cache)
+	return ociRemoteLoader{
+		cache:     cache,
+		dockerCli: dockerCli,
+	}, err
+}
+
+type ociRemoteLoader struct {
+	cache     string
+	dockerCli command.Cli
+}
+
+func (g ociRemoteLoader) Accept(path string) bool {
+	return strings.HasPrefix(path, "docker:")
```

Comment:
I believe elsewhere (e.g. build) we've used `docker-image://` as the format

Reply:
I guess this isn't technically an image, but a new artifact type, at which point I wonder if `oci://` would be better?

(I know flux2 uses that format for example: https://fluxcd.io/flux/cheatsheets/oci-artifacts/#consuming-artifacts)

Reply:
`oci` would be a better terminology from a pure implementation point of view, but I don't expect more than 1% Docker users know about Open Container Initiative, and just have a basic understanding of Docker images

Reply:
Definitely agree that most people don't know what OCI is 😛 

I would still say that:
1. We should at least use a URL scheme, i.e. `foo://bar` instead of `foo:bar`, being able to do a cheap check for `://` to distinguish between local & remote (without awareness of loader) is a helpful property
2. `docker` isn't really descriptive and could mean many different things/is un-Googleable, but "Docker Compose OCI include" will [hopefully] get results

Reply:
helm also uses `oci://` for modern helm push/pull